File: contains-dupes/contains_dupe.py
def containsDuplicate(numbers):

    # Approach 1
    unique = set()

    for num in numbers:
        if num in unique:
            return True
        else:
            unique.add(num)

    return False

    # Approach 2
    numbersSet = set(numbers)

    if len(numbers) != len(numbersSet):
        return True
    else:
        return False


result = containsDuplicate([9, 34, 67, 23, 3, 2, 3, 6])
print(result)

File: contains-dupes/contains_dupe.js
var containsDuplicate = function (nums) {
  // Approach 1
  unique = new Set();

  for (let i = 0; i < nums.length; i++) {
    if (unique.has(nums[i])) {
      return true;
    } else {
      unique.add(nums[i]);
    }
  }
  return false;

  // Approach 2
  uniqueSet = new Set(nums);
  return !(uniqueSet.size == nums.length);
};

console.log(containsDuplicate([9, 5, 3, 12, 6, 8, 1, 5]));
